resources/assets/lib/stores/channel-store.ts

```diff
@@ -230,6 +230,10 @@ export default class ChannelStore {
     }
 
     this.channels.delete(channelId);
+
+    if (this.channels.size <= 0) {
+      this.loaded = false;
+    }
```

Comment:
The fix should be done by fixing `MainView` to check channel size instead.

Reply:
So this should check the channel size instead of loaded?

https://github.com/ppy/osu-web/blob/ac2a44f6f13415a9a8d5db92c50297085734ede0/resources/assets/lib/chat/main-view.tsx#L37

Reply:
yes